libs/blocks/chart/chart.js

```diff
@@ -205,6 +206,68 @@ const areaSeriesOptions = (firstDataset) => (
   }))
 );
 
+export const setDonutLabel = (chart, label, unit = '', title = '') => {
+  chart.setOption({ series: [{ label: { formatter: [`{a|${label?.toLocaleString()}${unit}}`, `{b|${title}}`].join('\n') } }] });
+};
+
+export const handleDonutSelect = (source, selected, chart, unit, title) => {
+  const selectedSum = source.reduce((total, current) => {
+    if (selected[current[1]]) return total + current[0];
+    return total;
+  }, 0);
+  setDonutLabel(chart, selectedSum, unit, title);
+
+  return selectedSum;
+};
+
+export const donutSeriesOptions = (source, seriesData, size, unit, chart) => {
+  // Remove header names
+  source?.shift();
+  const sum = source?.reduce((total, current) => total + current[0], 0);
+  const firstSeries = seriesData?.[0];
+  const title = firstSeries ? propertyValueCI(firstSeries, 'title') : '';
+  const sizeLarge = size === LARGE;
+  let mouseOutValue = sum;
+
+  chart.on('mouseover', (value) => setDonutLabel(chart, value?.data?.[0], unit, title));
+  chart.on('mouseout', () => setDonutLabel(chart, mouseOutValue, unit, title));
+  chart.on('legendselectchanged', ({ selected }) => {
```

Comment:
Seem like a side effect to add these event listeners when getting the chart options, maybe that should be split out.